flake.nix

```diff
@@ -9,73 +9,22 @@
     flake-compat.inputs.nixpkgs.follows = "nixpkgs";
   };
 
-  # We use flake-parts as a way to make flakes 'system-aware'
-  # cf. https://github.com/NixOS/nix/issues/3843#issuecomment-661720562
   outputs = { self, nixpkgs, flake-parts, ... }:
     flake-parts.lib.mkFlake { inherit self; } {
       systems = nixpkgs.lib.systems.flakeExposed;
-      # The primed versions (self', inputs') are same as the non-primed
-      # versions, but with 'system' already applied.
-      perSystem = { self', inputs', pkgs, system, ... }:
-        let
-          inherit (pkgs.lib.lists) optionals;
-
-          # Specify GHC version here. To get the appropriate value, run:
-          #   nix-env -f "<nixpkgs>" -qaP -A haskell.compiler
-          hp = pkgs.haskellPackages; # Eg: pkgs.haskell.packages.ghc921;
-
-          # Specify your build/dev dependencies here.
-          shellDeps = with hp; [
-            cabal-fmt
-            cabal-install
-            ghcid
-            haskell-language-server
-            fourmolu
-            hlint
+      imports = [
+        ./haskell.nix
+      ];
+      perSystem = { pkgs, ... }: {
+        haskellProject = {
```

Comment:
I'm also wondering about naming convention here. I imagine a proliferation of `flake-parts` modules in the future, so the rust module for instance would look like:

```nix
inputs = {
  flake-part-rust = "github:rust-lang/flake-part-rust";
}
outputs = {
  ..
  imports = [
    flake-part-rust.flakeModule 
  ];
  perSystem = {
    rustProject = { root = ./.; };
  }
}
```

A few things to note:

- The flake-part repo itself is named `flake-part-foo` where "foo" is the feature it provides
- The module settings are named in camelcase (`rustProject`, or `haskellProject`) but this seems more arbitrary. Can the naming be improved?

Reply:
This looks fine to me and in accordance with the existing Nixpkgs naming conventions: camelCase for everything except snake-case for packages.



> So, should we call it `flake-foo` or `foo-flake` instead from the get go?

I honestly don't know. If the idea has enough momentum, many modules can be `flakeModule` attrs on the upstream projects, like I'm proposing here https://github.com/cachix/pre-commit-hooks.nix/pull/130.
Integrations can be `a.flakeModules.b` or `b.flakeModules.a`, depending on which upstream wants to accept the integration. The remainder will be in their own repos. `flake-part-foo`, `flake-foo` and `foo` may all be valid choices.